src/hypothesis/stateful.py

```diff
@@ -518,3 +575,8 @@ def execute_step(self, step):
             )
             for target in rule.targets:
                 self.bundle(target).append(VarReference(name))
+
+        for invar in self.invariants():
+            if invar.precondition is not None and not invar.precondition(self):
```

Comment:
As a trivial comment this probably reads more naturally as:

```python
if invar.precondition is not None and invar.precondition(self):
    invar.function(self)
```

but it doesn't really matter and I don't really care much either way :-) feel free to leave it as is.